cmd/controller/cmd/root.go

```diff
@@ -103,12 +105,38 @@ func executeRootCmd(baseCtx context.Context, cfg *config2.Config) error {
 	// set up signals so we handle the first shutdown signal gracefully
 	ctx := signals.SetupSignalHandler(baseCtx)
 
-	go func() {
-		err := profutils.StartProfilingServerWithDefaultHandlers(ctx, cfg.ProfilerPort.Port, nil)
+	// Add the propeller subscope because the MetricsPrefix only has "flyte:" to get uniform collection of metrics.
+	propellerScope := promutils.NewScope(cfg.MetricsPrefix).NewSubScope("propeller").NewSubScope(cfg.LimitNamespace)
+	mgr, err := controller.CreateControllerManager(ctx, cfg, defaultNamespace, &propellerScope)
+	if err != nil {
+		logger.Fatalf(ctx, "Failed to create controller manager. Error: %v", err)
+		return err
+	}
+
+	g, childCtx := errgroup.WithContext(ctx)
+	g.Go(func() error {
+		err := profutils.StartProfilingServerWithDefaultHandlers(childCtx, cfg.ProfilerPort.Port, nil)
+		if err != nil {
+			logger.Fatalf(childCtx, "Failed to Start profiling and metrics server. Error: %v", err)
+		}
+		return err
+	})
+
+	g.Go(func() error {
+		err := controller.StartControllerManager(childCtx, mgr)
```

Comment:
do we have to start this in a go-routine. Earlier we were not doing so?

Reply:
We also started it in go-routine before. https://github.com/flyteorg/flytepropeller/pull/421/files#diff-243ebed2765f75e6a54f57167212fefb08c3b2a85967ad2acbc0eb78919019c1L556-L564